RevenueCatUI/CustomerCenter/Data/PurchaseInformation.swift

```diff
@@ -92,23 +105,29 @@ struct PurchaseInformation {
         self.isCancelled = isCancelled
         self.latestPurchaseDate = latestPurchaseDate
         self.customerInfoRequestedDate = customerInfoRequestedDate
+        self.introductoryDiscount = introductoryDiscount
+        self.expirationDate = expirationDate
+        self.renewalDate = renewalDate
+        self.dateFormatter = Self.defaultDateFormatter
         self.managePurchaseURL = managePurchaseURL
     }
 
     // swiftlint:disable:next function_body_length
     init(entitlement: EntitlementInfo? = nil,
-         subscribedProduct: StoreProduct? = nil,
+         storeProduct: StoreProduct? = nil,
          transaction: Transaction,
          renewalPrice: PriceDetails? = nil,
          customerInfoRequestedDate: Date,
-         dateFormatter: DateFormatter = DateFormatter(),
-         managePurchaseURL: URL?) {
+         managePurchaseURL: URL?,
+         dateFormatter: DateFormatter = Self.defaultDateFormatter
+    ) {
         dateFormatter.dateStyle = .medium
 
         // Title and duration from product if available
-        self.title = subscribedProduct?.localizedTitle
-        self.durationTitle = subscribedProduct?.subscriptionPeriod?.durationTitle
+        self.title = storeProduct?.localizedTitle
```

Comment:
I noticed this looks weird if not available:

<img width="387" alt="Screenshot 2025-05-07 at 18 05 17" src="https://github.com/user-attachments/assets/2301d096-4ac6-4fc2-9b78-e1b7d337602c" />

Reply:
Just added this https://github.com/RevenueCat/purchases-ios/pull/5066/commits/ebccbe1d25741a27bbffc18348feaa4c87282c44

`?? productIdentifier` in case title is not there

Reply:
hmm alternative would be to use the Display Name from the Transaction? The reason we didn't expose display name initially was because it could be internal... but `productIdentfier` is also internal... If we have to choose between two internal things, maybe we should display display name better

Reply:
Let's tackle this once we remove `RevenueCat.Transaction`